homeassistant/components/epion/sensor.py

```diff
@@ -0,0 +1,127 @@
+        # Relevant keys are: deviceId, deviceName, locationId, lastMeasurement, co2, temperature, humidity, pressure
+        entities.append(EpionSensor(epion_base, epion_device, "co2", hass))
+        entities.append(EpionSensor(epion_base, epion_device, "temperature", hass))
+        entities.append(EpionSensor(epion_base, epion_device, "humidity", hass))
+        entities.append(EpionSensor(epion_base, epion_device, "pressure", hass))
+
+    async_add_entities(entities)
+
+
+class EpionSensor(SensorEntity):
+    """Representation of an Epion Air sensor."""
+
+    def __init__(self, epion_base, epion_device, key, hass) -> None:
+        """Initialize an EpionSensor."""
+        self._epion_base = epion_base
+        self._epion_device = epion_device
+        self._measurement_key = key
+        self._last_value: float | None = self.extract_value()
+        self._display_name = ""
+        if key == "temperature":
+            self._attr_native_unit_of_measurement = UnitOfTemperature.CELSIUS
+            self._attr_device_class = SensorDeviceClass.TEMPERATURE
+            self._display_name = "Temperature"
+        elif key == "humidity":
+            self._attr_native_unit_of_measurement = PERCENTAGE
+            self._attr_device_class = SensorDeviceClass.HUMIDITY
+            self._display_name = "Humidity"
+        elif key == "pressure":
+            self._attr_native_unit_of_measurement = UnitOfPressure.HPA
+            self._attr_device_class = SensorDeviceClass.ATMOSPHERIC_PRESSURE
+            self._display_name = "Pressure"
+        else:
+            self._attr_native_unit_of_measurement = CONCENTRATION_PARTS_PER_MILLION
+            self._attr_device_class = SensorDeviceClass.CO2
+            self._display_name = "CO2"
+
+        self._attr_state_class = SensorStateClass.MEASUREMENT
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, epion_device["deviceId"])},
+            manufacturer="Epion",
+            name=epion_device["deviceName"],
+        )
+        self.entity_id = generate_entity_id(
+            "sensor.{}", f"{epion_device['deviceId']}_{key}", hass=hass
+        )
+
+    @property
+    def unique_id(self):
+        """Return the unique ID for this sensor."""
+        return f"{self._epion_device['deviceId']}_{self._measurement_key}"
+
+    @property
+    def name(self):
+        my_device_id = self._epion_device["deviceId"]
+        device_name = "Unknown"
+
+        if my_device_id not in self._epion_base.device_data:
+            device_name = self._epion_device.get("deviceName", my_device_id)
+        else:
+            my_device = self._epion_base.device_data[my_device_id]
+            device_name = my_device["deviceName"]
+        return f"{device_name} {self._display_name}"
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the value reported by the sensor."""
+        if self._last_value is not None:
+            return round(self._last_value, 1)
```

Comment:
Please don't round values, prefer to set `suggested_precision`

Reply:
Removed rounding and used suggested precision instead, with 0 or 1 decimal precision depending on the type of measurement.